reconcile/utils/terrascript_client.py

```diff
@@ -1549,6 +1549,74 @@ def populate_tf_resource_s3(self, resource, namespace_info):
                 sqs_identifier, **notification_values)
             tf_resources.append(notification_tf_resource)
 
+        event_notifications = common_values.get('event_notifications')
+        sns_notifications = []
+        sqs_notifications = []
+
+        for event_notification in event_notifications:
+            destination_type = event_notification['destination_type']
+            destination_identifier = event_notification['destination']
+            event_type = event_notification['event_type']
+
+            if destination_type == 'sns':
+                notification_type = 'topic'
+                resource_arn_data = 'data.aws_sns_topic'
+            elif destination_type == 'sqs':
+                notification_type = 'queue'
+                resource_arn_data = 'data.aws_sqs_queue'
+
+            if destination_identifier.startswith('arn:'):
+                resource_name = destination_identifier.split(':')[-1]
+                resource_arn = destination_identifier
+            else:
+                resource_name = destination_identifier
+                resource_arn = '${'+resource_arn_data+'.' \
+                    + destination_identifier + '.arn}'
+
+            notification_config = {
+                'id': resource_name,
+                notification_type+'_arn': resource_arn,
+                'events': event_type
+            }
+
+            filter_prefix = event_notification.get('filter_prefix', None)
+            if filter_prefix is not None:
+                notification_config['filter_prefix'] = filter_prefix
+            filter_suffix = event_notification.get('filter_suffix', None)
+            if filter_suffix is not None:
+                notification_config['filter_suffix'] = filter_suffix
+
+            resource_values = {
+                'name': resource_name
+            }
+            resource_provider = values.get('provider')
+            if resource_provider:
+                resource_values['provider'] = resource_provider
+
+            if destination_type == 'sns':
+                resource_data = data.aws_sns_topic(resource_name,
```

Comment:
I think we need to add resource_data only if destination_identifier does not startswith('arn:'), otherwise terraform can not find the target resource.